### ISSUE TITLE
Notification.vue accessibility

### ISSUE THREAD

**Issue by williamschen23** [Submitty/Submitty#12340]
**Describe the bug**

Sometimes the system cypress test fails due to some accessibility issues in Notification.vue. 
```The element “a” must not appear as a descendant of an element with the attribute “role=link”.```
When there is a primary button, there cannot be nested buttons within it. The recommended approach, i think, is to have buttons be siblings of the same parent element, but use css to align them in such a way that they appear in the same order they do now.

**Expected behavior**

We should adhere to all accessibility guidelines

**To Reproduce**
https://github.com/Submitty/Submitty/actions/runs/21097673489/job/60681603352?pr=12339
This is an example of the test failing, even though it should realistically fail on every pr.
related to pr #12339
